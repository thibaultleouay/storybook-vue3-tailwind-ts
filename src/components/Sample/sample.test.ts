import { expect, test } from 'vitest'
import { mount } from '@vue/test-utils'

import Sample from './Sample.vue'

test('mount component', async() => {
  expect(Sample).toBeTruthy()

  const wrapper = mount(Sample, {
    props: {
      someProp: 'test',
      otherProp: 12,
    },
  })
  expect(wrapper.text()).toContain('test')
  expect(wrapper.html()).toMatchSnapshot()

  expect(wrapper.text()).toContain('12')
})
