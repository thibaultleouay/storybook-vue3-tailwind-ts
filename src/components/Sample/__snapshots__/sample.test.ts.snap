// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"border-8 border-red-300\\">
  <h1>Example component with VUE 3 and TS AND TAILWIND</h1>
  <p class=\\"text-3xl font-bold underline\\"> someProp has value of test</p>
  <p class=\\"text-3xl font-bold underline\\"> otherProp has value of 12</p>
</div>"
`;
